main/src/cgeo/geocaching/models/Geocache.java

```diff
@@ -1654,10 +1654,10 @@ public boolean addWaypointsFromText(@Nullable final String text, final boolean u
 
     private Waypoint findWaypoint(final Waypoint searchWp) {
```

Comment:
You could optimize the code by annotate with Nullable/NonNull here:
~~~
    @Nullable
    private Waypoint findWaypoint(@NonNull final Waypoint searchWp) {
~~~

Reply:
done